sktime/transformations/series/detrend/_deseasonalize.py

```diff
@@ -25,13 +23,39 @@
 class Deseasonalizer(_SeriesToSeriesTransformer):
     """A transformer that removes seasonal components from time series.
 
+    Fit computes seasonal components and stores them in `_seasonal`.
+
+    Transform aligns seasonal components stored in `_seasonal` with
+    the time index of the passed series and then
+    substracts them ("additive" model) from the passed series
+    or divides the passed series by them ("multiplicative" model).
+
     Parameters
     ----------
-    sp : int, optional (default=1)
+    sp : int, default=1
         Seasonal periodicity
-    model : str {"additive", "multiplicative"}, optional (default="additive")
+    model : str {"additive", "multiplicative"}, default="additive"
```

Comment:
I think you can remove the "str" part, since {"additive", "multiplicative"} indicates that only those two strings can be passed.